source/core/GenericFunctions.cpp

```diff
@@ -2232,7 +2269,9 @@ VIREO_FUNCTION_SIGNATURET(VectorScalarBinaryOp, AggregateBinOpInstruction)
     IntIndex count = srcArray1->Length();
 
     // Resize output to size of input array
-    destArray->Resize1D(count);
+    IntIndex* newDimensionLengths = srcArray1->DimensionLengths();
+    IntIndex rank = srcArray1->Rank();
+    destArray->ResizeDimensions(rank, newDimensionLengths, true);
```

Comment:
Looks like this was an oversight in the original as well, which means there probably isn't a test for this.
ReOrImToPolar or PolarToReOrIm on arrays should be a good test case for this,
and ComplexToReOrIm a test case for the un-op version.